code/permuter/Permuter.cpp

```diff
@@ -58,7 +58,7 @@ bool Permuter::FindOverlapsAndInsert(disk_write &dw,
     list<pair<unsigned int, unsigned int>> &ranges) const {
 
   unsigned int start = dw.metadata.write_sector;
-  unsigned int end = start + dw.metadata.size;
+  unsigned int end = start + dw.metadata.size - 1;
```

Comment:
Was this off by one bug? Does the list from 0 to N-1?

Reply:
yes, simple off by one bug. I tend to do math ranges and not computer science ranges when I'm on autopilot... but I guess that's what tests are for!